neo/app/actor/NeoActor.scala

```diff
@@ -1,57 +1,88 @@
 package actor
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
-import akka.pattern.{ask, pipe}
-import akka.util.Timeout
-import edu.uci.ics.cloudberry.zion.actor.BerryClient
+import akka.stream.Materializer
+import akka.util._
+import akka.stream.scaladsl._
+import edu.uci.ics.cloudberry.zion.common.Config
 import edu.uci.ics.cloudberry.zion.model.schema.TimeField
 import models.{GeoLevel, UserRequest}
 import org.joda.time.DateTime
+import play.api.Logger
 import play.api.libs.json._
+import play.api.libs.ws._
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
 
-class NeoActor(out: ActorRef, val berryClientProps: Props)(implicit ec: ExecutionContext) extends Actor with ActorLogging {
-
-  val berryClient = context.watch(context.actorOf(berryClientProps))
+class NeoActor(out: ActorRef, ws: WSClient, host: String, config: Config)
+              (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
 
   import NeoActor._
-
   implicit val timeout: Timeout = Timeout(20.minutes)
 
   override def receive: Receive = {
     case json: JsValue =>
       json.validate[UserRequest].map { userRequest =>
-        val allRequest = generateCBerryRequest(userRequest)
+
+        // Generate json in berry format
         import RequestType._
+        val berryRequest = generateCBerryRequest(userRequest)
+
+        // Get Berry url address
+        val url = "http://" + host + "/berry"
 
-        // Example of sending the non-slicing query using ask pattern
-        val sampleJson = allRequest(Sample)
-        val fAnswer = (berryClient ? sampleJson).mapTo[JsValue].map(json => Json.obj("key" -> "sample", "value" -> json))
-        fAnswer pipeTo out
+        // Dealing with response on samples: slicing
+        handleHttpTransport(url, "sample", berryRequest(Sample))
```

Comment:
in order to remove the above comment, you can write a function
`handleSamplingResponse`
later you can have another function `handleSlicingResponse`. Then people will know what you are doing.